packages/shared/src/components/ShareBar.tsx

```diff
@@ -42,53 +53,155 @@ export default function ShareBar({ post }: { post: Post }): ReactElement {
+              </div>
+            }
+          >
+            Facebook
+          </Button>
         </SimpleTooltip>
-        <SimpleTooltip content="Share on Facebook">
-          <ColorfulShareButton
+
+        <SimpleTooltip content="Share on Twitter">
+          <Button
+            className="group flex-col mb-2 font-normal text-center text-salt-90 hover:bg-transparent border-0 share btn-tertiary typo-caption1"
             tag="a"
-            href={getFacebookShareLink(href)}
+            href={getTwitterShareLink(href, post.title)}
             target="_blank"
             rel="noopener"
-            onClick={() => onClick(ShareProvider.Facebook)}
-            icon={<FacebookIcon secondary />}
-            className="btn-tertiary"
-          />
+            onClick={() => onClick(ShareProvider.Twitter)}
+            textPosition="justify-start"
+            icon={
+              <div className="flex justify-center items-center mb-2 group-hover:bg-theme-hover rounded-xl">
+                <TwitterIcon secondary className="w-7 h-7 text-white" />
+              </div>
+            }
+          >
+            Twitter
+          </Button>
         </SimpleTooltip>
+
+        {squads
+          ?.filter(
+            (squadItem) =>
+              squadItem.active &&
+              verifyPermission(squadItem, SourcePermissions.Post),
+          )
+          ?.map((squad) => (
+            <SimpleTooltip content={`Share on ${squad.name}`} key={squad.id}>
+              <Button
+                className="group flex-col mb-2 font-normal text-center text-salt-90 hover:bg-transparent border-0 share btn-tertiary typo-caption1"
+                key={squad.id}
+                onClick={() => onShareToSquad(squad)}
+                textPosition="justify-between"
+                icon={
+                  <div className="flex justify-center items-center mb-2">
+                    <SourceProfilePicture
+                      source={squad}
+                      className="w-10 h-10 squad"
+                    />
+                  </div>
+                }
+              >
+                <div className="truncate">@{squad.handle}</div>
+              </Button>
+            </SimpleTooltip>
+          ))}
+
+        {hasSquadAccess && squads?.length === 0 && (
+          <SimpleTooltip content="Create new squad">
+            <Button
+              className="group flex-col mb-2 font-normal text-center text-salt-90 hover:bg-transparent border-0 share btn-tertiary typo-caption1"
+              onClick={() =>
+                openNewSquadModal({
+                  origin: Origin.Share,
+                  redirectAfterCreate: false,
+                })
+              }
+              textPosition="justify-between"
+              icon={
+                <div className="flex justify-center items-center mb-2">
+                  <div className="flex justify-center items-center w-10 h-10 rounded-full squad bg-theme-color-cabbage">
```

Comment:
It shared the squad class from above?
Not sure this is then a generic component?